spec/lib/traffic_source_parser/parser/utmz_parser_spec.rb

```diff
@@ -18,15 +18,45 @@
         it_behaves_like 'a traffic source campaign parser'
       end
 
-      context 'when value is from social' do
+      context 'when medium is display' do
+        let(:cookie) { ' 186095035.1415927951.1.1.utmcsr=google|utmgclid=CI_WsND2-MECFezm7Aod6DsASw|utmccn=scm_beta|utmcmd=display|utmcct=Display_integra' }
+        let(:campaign) { 'scm_beta' }
+        let(:medium) { 'display' }
+        let(:channel) { 'Display' }
+        let(:source) { 'google' }
+
+        it_behaves_like 'a traffic source campaign parser'
+      end
+
+      context 'when medium is banner' do
+        let(:cookie) { '190008937.1415911938.2.2.utmcsr=botao-post|utmccn=xundacampaign|utmcmd=banner' }
+        let(:campaign) { 'xundacampaign' }
+        let(:medium) { 'banner' }
+        let(:channel) { 'Display' }
+        let(:source) { 'botao-post' }
+
+        it_behaves_like 'a traffic source campaign parser'
+      end
+
+            context 'when medium is ppc but source is social' do
+        let(:cookie) { '78301271.1443725657.1.1.utmcsr=facebook|utmccn=fb_retargeting_1|utmcmd=ppc' }
+        let(:campaign) { 'fb_retargeting_1' }
+        let(:medium) { 'ppc' }
+        let(:channel) { 'Social' }
+        let(:source) { 'Facebook' }
+
+        it_behaves_like 'a traffic source campaign parser'
+      end
+
+      context 'when medium is from social' do
 
         context 'with content and from referral medium' do
-          let(:cookie) { '231152653.1432828491.1.1.utmcsr=t.co|utmccn=(referral)|utmcmd=referral|utmcct=/EFzCFawFrk' }
+          let(:cookie) { '231152653.1432828491.1.1.utmcsr=t.co|utmccn=(referral)|utmcmd=referral|utmcct=/teste' }
           let(:campaign) { '(referral)' }
           let(:medium) { 'referral' }
           let(:channel) { 'Social' }
           let(:source) { 'Twitter' }
-          let(:content) { '/EFzCFawFrk' }
+          let(:content) { '/teste' }
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.